From 7c87df9c159aa1d228f0d77b37942216cff34922 Mon Sep 17 00:00:00 2001
From: Jiri Kosina <jkosina@suse.cz>
Date: Tue, 24 May 2016 16:38:10 +0200
Subject: [PATCH] bcache: bch_writeback_thread() is not freezable
Git-commit: 7c87df9c159aa1d228f0d77b37942216cff34922
Patch-mainline: v4.7-rc1
References: bsc#1076110

bch_writeback_thread() is calling try_to_freeze(), but that's just an
expensive no-op given the fact that the thread is not marked freezable.

I/O helper kthreads, exactly such as the bcache writeback thread, actually
shouldn't be freezable, because they are potentially necessary for
finalizing the image write-out.

Signed-off-by: Jiri Kosina <jkosina@suse.cz>
Signed-off-by: Jens Axboe <axboe@fb.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 drivers/md/bcache/writeback.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/md/bcache/writeback.c
+++ b/drivers/md/bcache/writeback.c
@@ -12,7 +12,6 @@
 #include "writeback.h"
 
 #include <linux/delay.h>
-#include <linux/freezer.h>
 #include <linux/kthread.h>
 #include <trace/events/bcache.h>
 
@@ -225,7 +224,6 @@ static void read_dirty(struct cached_dev
 	 */
 
 	while (!kthread_should_stop()) {
-		try_to_freeze();
 
 		w = bch_keybuf_next(&dc->writeback_keys);
 		if (!w)
@@ -438,7 +436,6 @@ static int bch_writeback_thread(void *ar
 				return 0;
 			}
 
-			try_to_freeze();
 			schedule();
 			continue;
 		}
